modules/gapi/src/backends/ie/giebackend.cpp

```diff
@@ -900,11 +909,34 @@ struct Infer: public cv::detail::KernelTag {
                         // non-generic version for now:
                         // - assumes all inputs/outputs are always Mats
                         for (auto i : ade::util::iota(ctx->uu.params.num_in)) {
-                            // TODO: Ideally we shouldn't do SetBlob() but GetBlob() instead,
-                            // and redirect our data producers to this memory
-                            // (A memory dialog comes to the picture again)
-                            IE::Blob::Ptr this_blob = extractBlob(*ctx, i);
-                            req.SetBlob(ctx->uu.params.input_names[i], this_blob);
+                            if (ctx->uu.params.rctx != nullptr) {
+                                GAPI_Assert(ctx->inShape(i) == cv::GShape::GFRAME &&
+                                                "Callback feature is supported for MediaFrame only");
+                                cv::util::any any_blob_params = ctx->inFrame(i).blobParams();
+                                auto ie_core = cv::gimpl::ie::wrap::getCore();
+
+                                std::pair<InferenceEngine::TensorDesc,
+                                          InferenceEngine::ParamMap> blob_params;
+
+                                try {
+                                    blob_params = cv::util::any_cast<std::pair<InferenceEngine::TensorDesc,
+                                                                               InferenceEngine::ParamMap>>(any_blob_params);
+                                } catch (...) {
+                                    GAPI_Assert(false && "Incorrect type of blobParams: "
+                                                          "expected std::pair<InferenceEngine::TensorDesc,"
+                                                                             "InferenceEngine::ParamMap>");
+                                }
+
+                                IE::Blob::Ptr this_blob = ctx->uu.params.rctx->CreateBlob(blob_params.first,
+                                                                                          blob_params.second);
+                                req.SetBlob(ctx->uu.params.input_names[i], this_blob);
```

Comment:
It seems all these details should go to the `extractBlob` and this code should remain a two-liner with that `SetBlob` line.

Reply:
Done